src/Analyzer/ClassMethodAnalyzer.php

```diff
@@ -429,9 +424,35 @@ private function getDocReturnDeclaration(ClassMethod $method)
             $result = implode('|', $parsedComment['return']);
 
             return $result;
-        } else {
-            return ' ';
+        } elseif ($this->dependencyGraph !== null) {
+            /** @var Class_ $methodClass */
+            $methodClass = $method->getAttribute('parent');
+            if ($methodClass) {
+                $ancestors = [];
+                if (!empty($methodClass->extends)) {
```

Comment:
An interface analyzer also should have similar logic

Reply:
@vkublytskyi what interface analyzer do you mean?
This code analyze parent classes and implemented interfaces

Reply:
I assume `Magento\SemanticVersionChecker\Analyzer\InterfaceAnalyzer` should contain something similar to `foreach ($methodInterface->extends as $ancestor) { ... }`.

But probably this is not a really important case as in real life most probably no one will extend the interface and then redeclare one of its method even if this is possible.

Reply:
@vkublytskyi InterfaceAnalizer use the same ClassMethodAnalyzer as ClassAnalyzer. So no need any additional code
https://github.com/magento/magento-semver/blob/develop/src/Analyzer/ClassAnalyzer.php#L132
https://github.com/magento/magento-semver/blob/develop/src/Analyzer/InterfaceAnalyzer.php#L142

Reply:
@zakdma in this case you need to handle a case when `$methodClass->extends` is an array (see https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Node/Stmt/Interface_.php#L10)

Reply:
@vkublytskyi I made changes. Please see latest commit